src/js/components/DataTable/TableCell.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import { Tooltip } from 'react-tippy';
+
+const TableCell = ({
+  column, value, original, children, tooltip, tooltipLabel,
+}) => {
+  let cellElement = <div>{ children || value}</div>;
+
+  // interpolate string by inserting values with ":" at the beginning with values from the cell
+  if (column.link && typeof column.link === 'string') {
+    const href = column.link
+      .split('/')
+      .map(it => (it[0] === ':' ? original[`${it.substring(1)}`] : it))
+      .join('/');
```

Comment:
Isn't it a bit overthought? What about doing:
```
Cell: (row) => TableCell(row, {link: `/openboxes/order/show/${row.original.id}`}
```

so `TableCell` would get `row` and some `props` as params and inside props could be for example `link`?

Reply:
I did it for convenience to not write the whole callback `(row) => ....` and pass just the component and  links or tooltips as attributes above.
Good point on the nested values like `origin.id` but I think it can be easily fixed by using lodash get function
like:
`
    const href = column.link
      .split('/')
      .map(it => (it[0] === ':' ? _.get(it.substring(1)) : it))
      .join('/');
`